### PR TITLE
Bump toolchain and remove some dead code

### PR DESCRIPTION
**Pull request by shigoel**
### Description:

Bump toolchain, stop using deprecated functions, ignore unused variables, and clean up the `make` output.

### Testing:

`make tests` fails. Need to update ELFSage.

### License:

By submitting this pull request, I confirm that my contribution is
made under the terms of the Apache 2.0 license.

### CODE REVIEW

**Comment by shigoel** [comment]
Note to self: still need to replace `Lean.HashMap` with `Std.HashMap`.

**Comment by alexkeizer** [comment]
> Note to self: still need to replace `Lean.HashMap` with `Std.HashMap`.

I'm working on a refactor in that area, maybe best to hold off on this replacement for now, so as to avoid merge conflicts

> clean up the make output
❤️

File: Arm/BitVec.lean [hunk 1]
```diff
@@ -316,7 +316,10 @@ def reverse (x : BitVec n) : BitVec n :=
     else acc
   reverseTR x 0 $ BitVec.zero n
 
-example : reverse 0b11101#5 = 0b10111#5 := by rfl
+example : reverse 0b11101#5 = 0b10111#5 := by
+  -- (FIXME) With leanprover/lean4:nightly-2024-08-29, just `rfl` sufficed here.
+  simp [reverse, reverse.reverseTR]
+  rfl
```
[the remark this answers is not in the document]
Comment (shigoel) [reply Arm/BitVec.lean:322]: Yes, using `decide` did cause similar problems.

[the remark this answers is not in the document]
Comment (pennyannn) [reply Arm/BitVec.lean:322]: Yes, I remember running into problems with `decide` and had to use `native_decide` for some of the tests. The error would involve something about `well-founded recursion` not being supported. Is there something I could read about the difference between `well-founded recursion` vs `structural recursion`?

[the remark this answers is not in the document]
Comment (pennyannn) [reply Arm/BitVec.lean:322]: PR submitted at https://github.com/leanprover/LNSym/pull/158 for fixing this issue.

**Pull request merged**